pvlib/iam.py

```diff
@@ -940,3 +941,362 @@ def schlick_diffuse(surface_tilt):
+        target_params = {'a_r': optimize_result.x.item()}
+
+    elif target_name == "physical":
+        L, n = optimize_result.x
+        # have to unpack order because search order may be different
+        if L > n:
+            L, n = n, L
+        target_params = {'n': n, 'K': 4, 'L': L}
+
+    return target_params
+
+
+def convert(source_name, source_params, target_name, weight=_sin_weight,
+            fix_n=True, xtol=None):
+    """
+    Convert a source IAM model to a target IAM model.
+
+    Parameters
+    ----------
+    source_name : str
+        Name of the source model. Must be ``'ashrae'``, ``'martin_ruiz'``, or
+        ``'physical'``.
+
+    source_params : dict
+        A dictionary of parameters for the source model.
+
+            If source model is ``'ashrae'``, the dictionary must contain
+            the key ``'b'``.
+
+            If source model is ``'martin_ruiz'``, the dictionary must
+            contain the key ``'a_r'``.
+
+            If source model is ``'physical'``, the dictionary must
+            contain the keys ``'n'``, ``'K'``, and ``'L'``.
+
+    target_name : str
+        Name of the target model. Must be ``'ashrae'``, ``'martin_ruiz'``, or
+        ``'physical'``.
+
+    weight : function, optional
+        A single-argument function of AOI (degrees) that calculates weights for
+        the residuals between models. Must return a float or an array-like
+        object. The default weight function is :math:`f(aoi) = 1 - sin(aoi)`.
+
+    fix_n : bool, default True
+        A flag to determine which method is used when converting from the
+        ASHRAE model to the physical model.
+
+        When ``source_name`` is ``'ashrae'`` and ``target_name`` is
+        ``'physical'``, if `fix_n` is ``True``,
+        :py:func:`iam.convert` will fix ``n`` so that the returned physical
+        model has the same x-intercept as the inputted ASHRAE model.
+        Fixing ``n`` like this improves the fit of the conversion, but often
+        returns unrealistic values for the parameters of the physical model.
+        If more physically meaningful parameters are wanted,
+        set `fix_n` to False.
+
+    xtol : float, optional
+        Passed to scipy.optimize.minimize.
+
+    Returns
+    -------
+    dict
+        Parameters for the target model.
+
+            If target model is ``'ashrae'``, the dictionary will contain
+            the key ``'b'``.
+
+            If target model is ``'martin_ruiz'``, the dictionary will
+            contain the key ``'a_r'``.
+
+            If target model is ``'physical'``, the dictionary will
+            contain the keys ``'n'``, ``'K'``, and ``'L'``.
+
+    Note
+    ----
+    Target model parameters are determined by minimizing
+
+    .. math::
+
+        \\sum_{\\theta=0}^{90} weight \\left(\\theta \\right) \\times
+        \\| source \\left(\\theta \\right) - target \\left(\\theta \\right) \\|
+
+    The sum is over :math:`\\theta = 0, 1, 2, ..., 90`.
```

Comment:
Not quite true; the code actually uses `np.linspace(0, 90, 100)`

Reply:
Pedantic nitpick: `linspace(0, 90, 90)` produces `[0., 1.0112, 2.0225, ... , 87.9775, 88.9888, 90.]`.  To produce `theta = 0, 1, 2, ..., 90` as the docstring states, I think we need `linspace(0, 90, 91)`.

Reply:
Not a nit, rather, that's the intent, because it feels natural to step by 1.0000000000000...

Reply:
I agree it feels natural to step by unity.  I was trying to point out that the current code (`np.linspace(0, 90, 90)`) steps by 90/89 $\approx$ 1.011236.  To step by unity from 0 to 90 inclusive, it should be changed to `linspace(0, 90, 91)`.